arangod/Agency/Store.cpp

```diff
@@ -688,25 +690,29 @@ void Store::clear() {
 }
 
 /// Apply a request to my key value store
-Store& Store::operator=(VPackSlice const& slice) {
-  TRI_ASSERT(slice.isArray());
+Store& Store::operator=(VPackSlice const& s) {
+  TRI_ASSERT(s.isObject());
+  TRI_ASSERT(s.hasKey("readDB"));
+  auto const& slice = s.get("readDB");
   TRI_ASSERT(slice.length() == 4);
 
   MUTEX_LOCKER(storeLocker, _storeLock);
   _node.applies(slice[0]);
 
-  TRI_ASSERT(slice[1].isObject());
-  for (auto const& entry : VPackObjectIterator(slice[1])) {
-    if (entry.value.isNumber()) {
-      auto const& key = entry.key.copyString();
-      if (_node.has(key)) {
-        auto tp = TimePoint(std::chrono::seconds(entry.value.getNumber<int>()));
-        _node(key).timeToLive(tp);
-        _timeTable.emplace(std::pair<TimePoint, std::string>(tp, key));
+  if (s.hasKey("version")) {
```

Comment:
Please add a comment what we are doing here. We will all have forgotten in 2 weeks time.

Reply:
Of course